### PR TITLE
MergeCohortVcfs address edge cases in records merging, add output file name prefix

### PR DESCRIPTION
**Pull request by epiercehoffman**
### Changes

In `/src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py`:
* Address edge cases outlined in issue #15: check that chrom, chr2, and end2 are identical when merging records
* Also check SVLEN and STRANDS when merging records to avoid collapsing INSs that differ only in SVLEN, or INVs that differ only in STRANDS
* Add comments explaining de-duplication approach since it is not obvious at first why it is implemented this way, but attempting to change it revealed its logic
* When merging records, if the ALT field is different among otherwise identical records, keep the more informative one - ex. keep the record with ALT = `<INS:ME:ALU>` over just `<INS>`
* (Updated) Remove all but two dummy GTs from merged VCF (keep exactly one 0/1 GT so `svtk bedcluster` clusters only on coordinates not samples)

In `/wdl/MergeCohortVcfs.wdl`:
* Increase preemptible tries in MergeDepthVcfs from 0 to 3 by default
* Delete localized input files after use to help save disk space
* Add "cohort" string input and make it the prefix for all output files, so that files from multiple runs of this WDL from different cohorts/projects will be named differently and more easily identifiable

In `/test/module04/MergeCohortVcfs.test.json`:
* Add (required) "cohort" input
* Update sv-pipeline docker (note that commit hash on docker tag says 3ea8aa7 but is actually up to date with dd187ca, since it was rebuilt with --skip-base-image-build so the tag was not changed)

All other JSONS:
* Update sv-pipeline docker

### Testing

1. `/src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py` tested locally in docker container on 3 post-Module01 VCFs from different gnomAD v3 batches from different PESR callers. Output was verified to be the same as previously prior to the additional SVLEN and STRANDS checks. After those checks, the number of records increased by 0.14% to include INVs and INSs previously merged with other similar but non-identical records. Records were strictly added to the output; none were deleted or changed.
2. `/wdl/MergeCohortVcfs.wdl` was submitted to Cromwell with `/test/module04/MergeCohortVcfs.test.json` as input parameters and the job was successful. The job ID was: `df6b9d29-4496-4b37-a489-3ccce5025d01`
3. `/scripts/test/validate.sh` validated all WDLs/JSONs successfully

### CODE REVIEW

**Review by mwalker174 — APPROVED** [review]
Thank you @epiercehoffman! This is excellent. We should start using your PR description format going forward. I have a very minor question below about the ALT field prioritization check. 

I think your changes to the `sv-pipeline` docker are going to conflict with #54. I think it would be best to wait for that to be merged into master, then rebase and rebuild your docker.

File: src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py
Comment (epiercehoffman) [reply src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py:53]: Yes, the reason I included the check for <> is because of BNDs like this, in which `[CHR15:101980213[N` is the ALT field: 
```
chr1    10457   gnomAD-SV_v3_mc5b21_manta_chr1_chr1_1   N       [CHR15:101980213[N      .       PASS    END=101980223;SVTYPE=BND;CHR2=chr15;STRANDS=--;SVLEN=-1;ALGORITHMS=manta;MEMBERS=manta___14697_cvd_hgv__45f8c7_1;EVIDENCE=SR    GT
```

**Comment by epiercehoffman** [comment]
### Updates
After rebasing and more changes, including adding back two dummy genotypes in the merged VCF, the sv-pipeline docker was re-built (`eph_records_match_by_chrom_mergevcfs-4a380e0`) to incorporate the latest changes and the following testing was performed:
1. `/wdl/MergeCohortVcfs.wdl` was submitted to Cromshell with `/test/module04/MergeCohortVcfs.test.json` as input and was successful. The job ID was `5ec86c15-ba51-4f33-a005-8847fc03f47a`. The depth VCF output was checked visually (changes were expected), and the `master_cluster_dups.bed` output was compared with the previous version using `diff` to confirm that it was unchanged.
2. `/wdl/MergeCohortVcfs.wdl` was submitted to Cromshell with inputs from 3 GMKF batches, for a multi-batch test. The job finished successfully in under 15 minutes.

I will plan to re-update the dockers in the input JSONs right before merging, in case of further conflicts with other branches.

File: src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py
Comment (mwalker174) [inline src/sv-pipeline/04_variant_resolution/scripts/merge_vcfs.py:50]: Look at this again, I think we should add a little more description to what's going on here, ie the intention that we assume the more informative ALT contains more colons such as `INS:MEI` or `DUP:TANDEM`

**Review by mwalker174 — APPROVED** [review]
Looks good - I have two minor suggestions assuming the stress test goes well. You can just re-test with the smaller vcfs and forego re-running the stress test.

**Comment by epiercehoffman** [comment]
### More updates
Final clean-up, scaling, and response to PR review:
* Add detail to comment about keeping more informative ALT field
* Change grep to awk to avoid issues with sample IDs
* Update docker in MergeCohortVcfs test JSON
* Split cohort sort and combined into separate tasks
* Remove `time` calls used in scaling process
* Set RAM & disk defaults to optimal for full GMKF cohort

### Testing
* Full GMKF cohort ran through MergeCohortVcfs successfully (in about 20 minutes, for less than $0.01!) with new memory defaults. Cromwell job ID `2a9b9751-31a4-42b4-b299-ff3e6adcc108`

**Pull request merged**
